File: src/09.tree/41.94.ts
import { TreeNode } from "./01.100";

export function inorderTraversal(root: TreeNode | null): number[] {
    const ans: number[] = [];
    const dfs = (node: TreeNode | null) => {
        if (!node) {
            return;
        }

        dfs(node.left);
        ans.push(node.val);
        dfs(node.right);
    };

    dfs(root);

    return ans;
}

File: src/09.tree/41.94.go
package tree

func InorderTraversal(root *TreeNode) []int {
	var dfs func(node *TreeNode)
	ans := []int{}
	dfs = func(node *TreeNode) {
		if node == nil {
			return
		}

		dfs(node.Left)
		ans = append(ans, node.Val)
		dfs(node.Right)
	}
	dfs(root)
	return ans
}
